pkg/cluster/check/kubernetes.go

```diff
@@ -79,13 +79,17 @@ func K8sAllNodesReportedAssertion(ctx context.Context, cluster ClusterInfo) erro
 // K8sFullControlPlaneAssertion checks whether all the master nodes are k8s master nodes.
 //
 //nolint:gocyclo,cyclop
-func K8sFullControlPlaneAssertion(ctx context.Context, cluster ClusterInfo) error {
-	clientset, err := cluster.K8sClient(ctx)
+func K8sFullControlPlaneAssertion(ctx context.Context, cl ClusterInfo) error {
+	clientset, err := cl.K8sClient(ctx)
 	if err != nil {
 		return err
 	}
 
-	expectedNodes := append(cluster.NodesByType(machine.TypeInit), cluster.NodesByType(machine.TypeControlPlane)...)
+	expectedNodes := append(cl.NodesByType(machine.TypeInit), cl.NodesByType(machine.TypeControlPlane)...)
+	expectedNodeIPs := sliceCollect(expectedNodes, func(v cluster.NodeInfo) string {
+		return v.IPs[0].String()
```

Comment:
should we collect all addresses here as well as verify that `actualNodes` is a subset of `allExpectedNodeIPs` ?

in fact what we want to verify that every `Node` matches some `NodeInfo`, I wonder if that can be extracted? comparing list of addresses is a bit more weak check than matching `Node` and `NodeInfo`.

Reply:
I introduced a function, `assertNodes` which does assertion by comparing expectedNodes<>actualNodes, asserting that they are the same set of nodes (by their internalips) + for each node, asserting that the actual nodes IPs are a **subset** of the expected one. Can you have a look into that?